codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MessageBuilder.java

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright 2014 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.mqtt;
+
+import io.netty.buffer.Unpooled;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public final class MessageBuilder {
+
+    public static class PublishBuilder {
+        private String topic;
+        private boolean retained;
+        private MqttQoS qos;
+        private byte[] payload;
+        private int messageId;
+
+        public PublishBuilder topicName(String topic) {
+            this.topic = topic;
+            return PublishBuilder.this;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Useless qualified this usage in the same class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUselessQualifiedThis)